olive/cli/quantize.py

```diff
@@ -157,16 +212,119 @@ def run(self):
+        "onnx_model_defaults": {"implementation": "nvmo", "precision": "int4"},
+        "description": "(HfModel, OnnxModel) WOQ with AWQ.",
+    },
+    "gptq": {
+        "implementations": ["gptq", "quarot", "matmul4", "inc_static", "inc_dynamic"],
+        "hf_model_defaults": {"implementation": "gptq", "precision": "int4"},
+        "onnx_model_defaults": {"implementation": "matmul4", "precision": "int4"},
+        "description": "(HfModel, OnnxModel) WOQ with GPTQ.",
+    },
+    "rtn": {
+        "implementations": ["quarot", "bnb4", "matmul4", "nvmo"],
+        "hf_model_defaults": {"implementation": "quarot", "precision": "int16"},
+        "onnx_model_defaults": {"implementation": "onnx_static", "precision": "int8"},
+        "description": "(HfModel, OnnxModel) WOQ with RTN.",
+    },
+    "hqq": {
+        "implementations": ["matmul4"],
+        "hf_model_defaults": {"implementation": None, "precision": None},
+        "onnx_model_defaults": {"implementation": "matmul4", "precision": "int4"},
+        "description": "(OnnxModel) HQQ quantization using onnxruntime.",
+    },
+    "static": {
+        "implementations": ["onnx_static", "inc_static"],
+        "hf_model_defaults": {"implementation": None, "precision": None},
+        "onnx_model_defaults": {"implementation": "onnx_static", "precision": "int8"},
+        "description": "(OnnxModel) Static quantization using onnxruntime.",
+    },
+    "dynamic": {
+        "implementations": ["onnx_dynamic", "inc_dynamic"],
+        "hf_model_defaults": {"implementation": None, "precision": None},
+        "onnx_model_defaults": {"implementation": "onnx_dynamic", "precision": "int8"},
+        "description": "(OnnxModel) Dynamic quantization using onnxruntime.",
+    },
+}
+
+IMPLEMENTATIONS = {
+    "awq": {
+        "name": "WOQ with AWQ",
+        "supported_precisions": [],
+        "precision_mapping": {
+            "int4": 4,
+            "int8": 8,
+            "int16": 16,
+            "uint4": 4,
+            "uint8": 8,
+            "uint16": 16,
+        },
+    },
+    "gptq": {
+        "name": "WOQ with GPTQ",
+        "supported_precisions": [],
+        "precision_mapping": {
+            "int4": 4,
+            "int8": 8,
+            "int16": 16,
+            "uint4": 4,
+            "uint8": 8,
+            "uint16": 16,
+        },
+    },
+    "quarot": {
+        "name": "QuaRot/Hadamard rotation",
+        "supported_precisions": [],
+        "precision_mapping": {
+            "int4": 4,
+            "int8": 8,
+            "int16": 16,
+            "uint4": 4,
+            "uint8": 8,
+            "uint16": 16,
+        },
+    },
+    "bnb4": {
+        "name": "Bits-n-Bytes",
+        "supported_precisions": ["fp4", "nf4"],
+        "precision_mapping": {},
+    },
+    "matmul4": {
+        "name": "WOQ with MatMulNBits",
+        "supported_precisions": [],
+        "precision_mapping": {"fp32": 1, "fp16": 2, "bf16": 3, "int8": 4},
```

Comment:
matmul4 only supports int4. accuracy_level is not really a model precision, it is an internal compute detail https://github.com/microsoft/Olive/blob/d0d495c0fdf1c5f5ed6500953920a560aac2ea96/olive/passes/onnx/quantization.py#L622 

what happens when we run `olive quantize --implementation matmul4 --precision int4`? if it fails, it is not intuitive since the model precision is indeed int4. 
I think it might be better to expose the accuracy level directly using something like `--matmul4_accuracy_level` similar to `--quarot_rotate`.

Reply:
Removed the precision option for matmul and enforce only int4 quantization.